### PR TITLE
Three-center overlap integrals using Libint2

### PR DESCRIPTION
**Pull request by maxscheurer**
## Description
This PR refactors `ThreeCenterOverlapInt` using Libint2. Even though one cannot compute these integrals directly, it's possible to obtain them through the delta function: `<phi(r_1) phi(r_1)| delta(r_1 - r_2) | phi(r_2) 1>`.

## Todos

- [x] 3-center overlap integrals with L2

## Checklist
- [x] Tests added for any new features (SAPT-DFT still working, Python API via `MintsHelper` tested locally 👍)
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [ ] Ready for merge

### CODE REVIEW

**Review by jturney — APPROVED** [review]
Nice!

**Review by JonathonMisiewicz — APPROVED** [review]

**Review by loriab — APPROVED** [review]
nice negative loc!

**Comment by maxscheurer** [comment]
Let's wait for @andysim's review until merging 😄

**Review by andysim — APPROVED** [review]
Excellent stuff.  A lot of negative lines of code, with many more to come in osrecur.cc eventually :)

**Pull request merged**
